tests/components/tasmota/test_binary_sensor.py

```diff
@@ -0,0 +1,222 @@
+"""The tests for the Tasmota binary sensor platform."""
+import copy
+from datetime import timedelta
+import json
+
+from hatasmota.utils import (
+    get_topic_stat_status,
+    get_topic_stat_switch,
+    get_topic_tele_sensor,
+    get_topic_tele_will,
+)
+
+from homeassistant.components import binary_sensor
+from homeassistant.components.tasmota.const import DEFAULT_PREFIX
+from homeassistant.const import (
+    ATTR_ASSUMED_STATE,
+    EVENT_STATE_CHANGED,
+    STATE_OFF,
+    STATE_ON,
+)
+import homeassistant.core as ha
+import homeassistant.util.dt as dt_util
+
+from .test_common import (
+    DEFAULT_CONFIG,
+    help_test_availability,
+    help_test_availability_discovery_update,
+    help_test_availability_when_connection_lost,
+    help_test_discovery_device_remove,
+    help_test_discovery_removal,
+    help_test_discovery_update_unchanged,
+    help_test_entity_id_update_discovery_update,
+    help_test_entity_id_update_subscriptions,
+)
+
+from tests.async_mock import patch
+from tests.common import async_fire_mqtt_message, async_fire_time_changed
+
+
+async def test_controlling_state_via_mqtt(hass, mqtt_mock, setup_tasmota):
+    """Test state update via MQTT."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["swc"][0] = 1
+    mac = config["mac"]
+
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.test")
+    assert state.state == "unavailable"
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/LWT", "Online")
+    state = hass.states.get("binary_sensor.test")
+    assert state.state == STATE_OFF
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6299130)